lib/auth/people_roles.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Auth.PeopleRoles do
+  @moduledoc """
+  Defines people_roles schema and fuction to grant roles to a person.
+  """
+  use Ecto.Schema
+  import Ecto.Changeset
+  alias Auth.Repo
+  # https://stackoverflow.com/a/47501059/1148249
+  alias __MODULE__
+
+  schema "people_roles" do
+    belongs_to :person, Auth.Person
+    belongs_to :role, Auth.Role
+    field :granter_id, :integer
+
+    timestamps()
+  end
+
+  @doc """
+  grant_role/3 grants a role to the given person
+  the conn must have conn.assigns.person to check for admin in order to grant the role.
+  """
+  def grant_role(conn, grantee_id, role_id) do
```

Comment:
Does `conn` need to be passed to be passed here, as opposed to the `Person` struct directly?

Passing conn here will mix up the M and C in `MVC` where the Model should only be concerned with the applications data structures and not the current web session

Reply:
@th0mas good question. I thought about that too. 💭 
Ultimately I want to have a means of redirecting an unauthorised request. 
So what I might end up doing is create a function in the `role_controller.ex`
and just transform this `grant_role/3` to be a simpler insert  function. 👍

Reply:
Hi @th0mas, as discussed/suggested, I've refactored the function to:

https://github.com/dwyl/auth/blob/d3fa7de6e93a82d077b2e959b31f52ac95f42950/lib/auth/people_roles.ex#L28-L40

And updated the corresponding test. 👍 

https://github.com/dwyl/auth/blob/d3fa7de6e93a82d077b2e959b31f52ac95f42950/test/auth/people_roles_test.exs#L4-L19

Always happy to get your feedback on things. 🙌